src/qofonomodem.cpp

```diff
@@ -24,9 +25,21 @@ Q_GLOBAL_STATIC(ModemMap, modemMap)
 
 #define SUPER QOfonoObject
 
+class QOfonoModem::Private : public SUPER::ExtData
+{
+public:
+    bool modemPathValid;
+    QSharedPointer<QOfonoManager> mgr;
+    Private() : modemPathValid(false), mgr(QOfonoManager::instance()) {}
+};
+
 QOfonoModem::QOfonoModem(QObject *parent) :
-    SUPER(parent)
+    SUPER(new Private, parent)
 {
+    QOfonoManager* mgr = privateData()->mgr.data();
+    connect(mgr, SIGNAL(availableChanged(bool)), SLOT(onManagerAvailableChanged(bool)));
+    connect(mgr, SIGNAL(modemsChanged(QStringList)), SLOT(onModemsChanged(QStringList)));
```

Comment:
There is no need to define these two private slots which just call checkModemPathValidity(). The slots can have fewer parameters than the signal. Make checkModemPathValidity() a slot and connect to the signals directy.